File: java/scala/src/main/scala/com/asiafrank/scala/bean/Child.scala
package com.asiafrank.scala.bean

import scala.beans.BeanProperty

/**
  * Child
  *
  * Created at 2017/2/8 0008.
  *
  * @author zhangxf
  */
class Child {
  @BeanProperty
  var id: Long = _

  @BeanProperty
  var name: String = _

  override def toString = s"Child($id, $name)"
}

File: java/akka-java/src/main/java/com/asiafrank/akka/fault/example2/Child.java
package com.asiafrank.akka.fault.example2;

import akka.actor.UntypedActor;

/**
 * Child
 * <p>
 * </p>
 * Created at 1/9/2017.
 *
 * @author zhangxf
 */
final class Child extends UntypedActor {
    int state = 0;

    @Override
    public void onReceive(Object message) throws Throwable {
        if (message instanceof Exception) {
            throw (Exception) message;
        } else if (message instanceof  Integer) {
            state = (Integer) message;
        } else if (message.equals("get")) {
            getSender().tell(state, getSelf());
        } else {
            unhandled(message);
        }
    }
}
